### PR TITLE
change the PARSE_TOKEN to skipParseTokenEnv and fix  the isTokenExpired logic issue 

### PR DESCRIPTION
**Pull request by williamaronli**
ref: https://github.com/istio/istio/issues/26225

The root cause: when it needs to rotate the cert, token expiration are not validated as the logic is not set correctly in the isTokenExpired  function: 

1. k8s environment: the default value skipParseTokenEnv should be true. So it will not skip the token expire checking here
https://github.com/ist
io/istio/blob/master/security/pkg/nodeagent/cache/secretcache.go#L946

2. vm environment: skipParseTokenEnv should be set to false. As when rotate it should use cert instead of token

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by jwendell — COMMENTED** [review]
Does this fix https://github.com/istio/istio/issues/26225? Can we have a unit or integration test that validates that?

**Comment by howardjohn** [comment]
We REALLY need a test for this. We cannot have the most basic, core authentication feature be broken

**Review by costinm — COMMENTED** [review]
I'm afraid this is just masking other problems - and likely will still break in some environments.

The prov cert should be used in all cases - if it exists. That's consistent with previous behavior - when connecting to XDS, if /etc/certs - or the new equivalents - exist we use them.

If we have to use the token - either because there is no cert, or we're getting a new workload cert -
the token should be loaded from the file and not cached.  

Parsing the token is a very weird thing - I'm not aware of anyone else doing this, usually it's not even possible since some tokens are opaque strings. For JWT we can parse - but that's not something we should depend on,

**Comment by costinm** [comment]
For test - I don't see how we can correctly test this without a long-running test, maybe with a shortened expiration for the JWT token mounted by k8s. Stability seems the best way to catch cert rotation and JWT rotation, and no amount of unit tests or short lived integrations will fully cover this critical functionality.

**Comment by williamaronli** [comment]
> I'm afraid this is just masking other problems - and likely will still break in some environments.
> 
> The prov cert should be used in all cases - if it exists. That's consistent with previous behavior - when connecting to XDS, if /etc/certs - or the new equivalents - exist we use them.
> 
> If we have to use the token - either because there is no cert, or we're getting a new workload cert -
> the token should be loaded from the file and not cached.
> 
> Parsing the token is a very weird thing - I'm not aware of anyone else doing this, usually it's not even possible since some tokens are opaque strings. For JWT we can parse - but that's not something we should depend on,

we will change the name of PARSE_TOKEN to SkipParseToken, and set he default value to false. and the SkipParseToken variable will not be changed depending on PROV_CERT value.

I think this is more user friendly. and the code more readability and easy to maintain.

As user wants to use cert for vm, only set the SkipParseToken to True.
if user wants to use k8s, user don't need to set the SkipParseToken.

there are few advantages:
1. easy to maintain the code. the environment  variable should not be changed when it is set.
2. user friendly. only needs to control  the SkipParseToken env variable to choose to use vm or token

**Comment by liminw** [comment]
> I'm afraid this is just masking other problems - and likely will still break in some environments.
> 
> The prov cert should be used in all cases - if it exists. That's consistent with previous behavior - when connecting to XDS, if /etc/certs - or the new equivalents - exist we use them.
> 
> If we have to use the token - either because there is no cert, or we're getting a new workload cert -
> the token should be loaded from the file and not cached.
> 
> Parsing the token is a very weird thing - I'm not aware of anyone else doing this, usually it's not even possible since some tokens are opaque strings. For JWT we can parse - but that's not something we should depend on,

Parse_token here is for checking expiration. The long term fix, I think, should always fetch a new token (from file, or through credential fetcher) when we need to refresh cert (which is very infrequent). This way, we can remove the parsing token flag. 

For 1.7, we should leave it to the current behavior to avoid any surprise.

**Review by liminw — APPROVED** [review]
/lgtm

File: pilot/cmd/pilot-agent/main.go [hunk 1]
```diff
@@ -144,8 +144,8 @@ var (
 	useTokenForCSREnv   = env.RegisterBoolVar("USE_TOKEN_FOR_CSR", false, "CSR requires a token").Get()
 	credFetcherTypeEnv  = env.RegisterStringVar("CREDENTIAL_FETCHER_TYPE", "",
 		"The type of the credential fetcher. Currently supported types include GoogleComputeEngine").Get()
-	parseTokenEnv = env.RegisterBoolVar("PARSE_TOKEN", false,
-		"Parse token to inspect information like expiration time in proxy. This may not always be possible because token may not be a JWT.").Get()
+	skipParseTokenEnv = env.RegisterBoolVar("SKIP_PARSE_TOKEN", false,
+		"Skip Parse token to inspect information like expiration time in proxy. This may not always be possible because token may not be a JWT.").Get()
```
Comment (JimmyCYJ) [inline pilot/cmd/pilot-agent/main.go:148]: "Skip parse token", could you also rephrase the flag description?

Comment (williamaronli) [reply pilot/cmd/pilot-agent/main.go]: updated it

**Review by JimmyCYJ — APPROVED** [review]

**Pull request merged**